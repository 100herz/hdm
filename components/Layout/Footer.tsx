import Link from "next/link"
import styled from "styled-components"

const Address = styled.address`
  font-size: 1.125rem;
  font-style: normal;
  line-height: 2.5rem;
`

type Props = {
  className?: string
}

const Footer: React.FC<Props> = ({ className = "" }: Props) => {
  return (
    <footer className={`container px-6  ${className}`}>
      <section>
        <h2 className="py-16 text-4xl font-bold leading-10">
          Nehmen Sie <br />
          Kontakt mit <br />
          uns auf
        </h2>

        <Address>
          Holz Design Moers GmbH <br />
          Am Schürmannshütt 23 <br />
          47441 Moers
        </Address>

        <Address className="mt-6">
          Telefon:{" "}
          <a className="underline" href="tel:+49284114010">
            +49 2841 1401 0
          </a>{" "}
          <br />
          Telefax: +49 2841 1401 180 <br />
          E-Mail:{" "}
          <a className="underline" href="mailto:info@hdm.gmbh">
            info@hdm.gmbh
          </a>
        </Address>

        <div className="mt-24">
          <img className="w-48" src="/ci/hdm-logo.svg" alt="Logo der HDM GmbH" />
        </div>
      </section>

      <section className="mt-6 text-lg leading-10">
        © 2020 Holz Design Moers GmbH
        <nav className="mt-2">
          <Link href="/imprint" as="impressum">
            <a>Impressum</a>
          </Link>
          <Link href="/privacy" as="datenschutz">
            <a className="ml-8">Datenschutz</a>
          </Link>
        </nav>
      </section>

      <div className="mt-10 text-right">
        <button>
          <svg
            className="inline-block w-6 transform -rotate-90"
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 24.242 45"
          >
            <path
              d="M126.031,21.266,105.252.486A1.732,1.732,0,0,0,102.8,2.935L122.359,22.49,102.8,42.045a1.731,1.731,0,0,0,2.448,2.448l20.779-20.779A1.732,1.732,0,0,0,126.031,21.266Z"
              transform="translate(-102.297 0)"
              fill="#0084b0"
            />
          </svg>
        </button>
      </div>
    </footer>
  )
}

export default Footer
