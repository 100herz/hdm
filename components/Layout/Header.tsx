import Link from "next/link"
import styled from "styled-components"

const Section = styled.section`
  position: relative;

  &::before {
    position: absolute;
    top: 0;
    left: 0;
    width: calc(2.9333 * 90%);
    height: 90%;
    background: url(/ci/hdm-bg.svg) no-repeat;
    opacity: 0.05;
    content: "";
  }

  @media (min-width: 768px) {
    position: static;

    &::before {
      left: 6rem;
      width: calc(2.9333 * 100%);
      height: 100%;
    }
  }
`

type Props = {
  className?: string
}

const Header: React.FC<Props> = ({ className = "" }: Props) => {
  return (
    <header className={`bg-gray-100 relative overflow-x-hidden ${className}`}>
      <div className="container justify-between my-6 lg:flex">
        <h1 className="pt-8">
          <img className="w-56 mx-auto" src="/ci/hdm-logo.svg" alt="Logo der HDM GmbH" />
          <span className="sr-only">HDM - Holz Design Moers GmbH</span>
        </h1>

        <nav className="flex justify-between my-16 font-semibold lg:text-lg lg:container lg:mb-0 lg:space-x-12 lg:justify-end">
          <Link href="/#produkte">
            <a>Produkte</a>
          </Link>
          <Link href="/#unternehmen">
            <a>Unternehmen</a>
          </Link>
          <Link href="/#kontakt">
            <a>Kontakt</a>
          </Link>
        </nav>
      </div>

      <Section className="container pt-12 pb-20 text-xl leading-10 lg:text-3xl lg:py-40">
        <p className="lg:max-w-3xl lg:ml-24">
          <strong>Produkt. Idee. Erfolg.</strong> Die Holz Design Moers GmbH bietet ihren Handelspartnern auch in
          Zukunft abverkaufsstarke Produkte, Konzepte und Ideen für eine erfolgreiche Zusammenarbeit.
        </p>
      </Section>

      <nav className="justify-around hidden px-20 my-8 text-lg font-semibold lg:flex">
        <Link href="/#crossflex">
          <a className="px-2">Crossflex</a>
        </Link>
        <Link href="/#purity">
          <a className="px-2">Purity</a>
        </Link>
        <Link href="/#cerastar">
          <a className="px-2 border-b-4 border-hdm">Cerastar</a>
        </Link>
        <Link href="/#massivholztueren">
          <a className="px-2">Massivholztüren</a>
        </Link>
        <Link href="/#moebelbauplatten">
          <a className="px-2">Möbelbauplatten</a>
        </Link>
        <Link href="/#dazu-zubehoer">
          <a className="px-2">Dazu Zubehör</a>
        </Link>
      </nav>
    </header>
  )
}

export default Header
