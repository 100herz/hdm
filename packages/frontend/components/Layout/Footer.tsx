import Link from 'next/link'
import Image from 'next/image'
import { useRouter } from 'next/router'
import { animateScroll } from 'react-scroll'
import styled from '@emotion/styled'

export default function Footer({ className = '' }: React.HTMLAttributes<HTMLElement>) {
  const router = useRouter()

  return (
    <FooterStyled id="kontakt" className={`lg:mt-40 pb-10 ${className}`}>
      <section className="container lg:flex lg:justify-between lg:space-x-6">
        <h2 className="py-16 text-4xl font-bold leading-10 lg:pt-0">
          Nehmen Sie <br />
          Kontakt mit <br />
          uns auf
        </h2>

        <Address>
          Holz Design Moers GmbH <br />
          Am Schürmannshütt 23 <br />
          47441 Moers
        </Address>

        <Address className="mt-6 lg:mt-0">
          Telefon:{' '}
          <a className="underline hover:text-hdm-dark" href="tel:+4928411401101">
            +49 2841 1401 101
          </a>
          <br />
          Telefax: +49 2841 1401 180 <br />
          E-Mail:{' '}
          <a className="underline hover:text-hdm-dark" href="mailto:info@hdm.gmbh">
            info@hdm.gmbh
          </a>
        </Address>

        <div className="mt-24 lg:mt-0">
          {router && router.pathname === '/' ? (
            <button
              className="w-48"
              id="logo"
              onClick={() => animateScroll.scrollToTop({ duration: 800 })}
              type="button"
              aria-label="HDM Logo"
            >
              <Image src="/ci/hdm-logo.svg" alt="Logo der HDM GmbH" width={192} height={49} />
            </button>
          ) : (
            <Link href="/">
              <a className="w-48">
                <Image src="/ci/hdm-logo.svg" alt="Logo der HDM GmbH" width={192} height={49} />
              </a>
            </Link>
          )}
        </div>
      </section>

      <section className="container mt-6 text-lg leading-10 lg:flex lg:justify-center">
        <p>&copy; 2020 Holz Design Moers GmbH</p>
        <nav className="flex flex-col mt-2 leading-8 lg:mt-0 lg:flex-row">
          <Link href="/downloads">
            <a className="underline lg:ml-8 hover:text-hdm-dark">Downloads</a>
          </Link>
          <Link href="/impressum">
            <a className="underline lg:ml-8 hover:text-hdm-dark">Impressum</a>
          </Link>
          <Link href="/datenschutz">
            <a className="underline lg:ml-8 hover:text-hdm-dark">Datenschutz</a>
          </Link>
        </nav>
      </section>

      <div className="container my-10 text-right lg:text-center">
        <button
          type="button"
          id="back-to-top-button"
          onClick={() => animateScroll.scrollToTop({ duration: 800 })}
          aria-label="Go to top"
        >
          <svg
            className="inline-block w-6 transform -rotate-90"
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 24.242 45"
          >
            <path
              d="M126.031,21.266,105.252.486A1.732,1.732,0,0,0,102.8,2.935L122.359,22.49,102.8,42.045a1.731,1.731,0,0,0,2.448,2.448l20.779-20.779A1.732,1.732,0,0,0,126.031,21.266Z"
              transform="translate(-102.297 0)"
              fill="#0084b0"
            />
          </svg>
        </button>
      </div>
    </FooterStyled>
  )
}

const FooterStyled = styled.footer`
  position: relative;

  @media (min-width: 1024px) {
    background-image: linear-gradient(rgba(255, 255, 255, 0.95), rgba(255, 255, 255, 0.95)), url(/ci/hdm-bg.svg);
    background-repeat: no-repeat;
    background-size: 100% auto;
  }
`

const Address = styled.address`
  font-size: 1.125rem;
  font-style: normal;
  line-height: 2.5rem;
`
