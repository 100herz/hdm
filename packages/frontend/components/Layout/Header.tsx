import NavLink from 'next/link'
import Image from 'next/image'
import { useRouter } from 'next/router'
import { Link } from 'react-scroll'
import Sticky from 'react-stickynode'
import styled from '@emotion/styled'

import { useWindowDimensions } from '@/hooks/use-window-dimensions'

export default function Header({ className = '' }: React.HTMLAttributes<HTMLElement>) {
  const router = useRouter()
  const { width = 0 } = useWindowDimensions()

  return (
    <header className={`relative overflow-x-hidden ${className}`}>
      <div className="justify-between mt-6 lg:container lg:flex">
        <h1 className="pt-8 mb-10 lg:mb-0" data-testid="main-logo">
          <NavLink href="/">
            <a className="block w-56 mx-auto" aria-label="HDM - Holz Design Moers GmbH">
              <Image src="/ci/hdm-logo.svg" alt="Logo der HDM GmbH" width={224} height={57} />
            </a>
          </NavLink>
        </h1>

        <Sticky enabled={width < 1024} innerZ="101">
          {router && router.pathname === '/' ? (
            <nav
              className="flex justify-between py-3 font-semibold bg-white sm:justify-around lg:bg-transparent lg:text-lg lg:container lg:mb-0 lg:space-x-12 lg:justify-end lg:mt-10 lg:py-0"
              aria-label="Hauptnavigation auf der Startseite und in der mobilen Ansicht."
            >
              <Link
                to="produkte"
                activeClass="active"
                spy={true}
                smooth={true}
                offset={-120}
                duration={1000}
                className="px-4 py-2 my-2 hover:text-hdm"
              >
                Produkte
              </Link>
              <Link
                to="unternehmen"
                activeClass="active"
                spy={true}
                smooth={true}
                offset={-60}
                duration={1000}
                className="px-4 py-2 my-2 hover:text-hdm"
              >
                Unternehmen
              </Link>
              <Link
                to="kontakt"
                activeClass="active"
                spy={true}
                smooth={true}
                duration={1000}
                className="px-4 py-2 my-2 hover:text-hdm"
              >
                Kontakt
              </Link>
            </nav>
          ) : (
            <nav
              className="flex justify-between py-3 font-semibold bg-white lg:bg-transparent lg:text-lg lg:container lg:mb-0 lg:space-x-12 lg:justify-end lg:mt-10 lg:py-0"
              aria-label="Hauptnavigation nicht auf der Startseite aber in der mobilen Ansicht."
            >
              <NavLink href="/#produkte">
                <a className="px-4 py-2 my-2 hover:text-hdm">Produkte</a>
              </NavLink>
              <NavLink href="/#unternehmen">
                <a className="px-4 py-2 my-2 hover:text-hdm">Unternehmen</a>
              </NavLink>
              <NavLink href="/#kontakt">
                <a className="px-4 py-2 my-2 hover:text-hdm">Kontakt</a>
              </NavLink>
            </nav>
          )}
        </Sticky>
      </div>

      <Section className="container pt-12 pb-20 lg:flex lg:flex-row lg:justify-between lg:pt-32 lg:pb-24">
        <p className="text-xl leading-10 lg:text-3xl lg:max-w-3xl lg:ml-24">
          <strong>Produkt. Idee. Erfolg.</strong> Die Holz Design Moers GmbH bietet ihren Handelspartnern auch in
          Zukunft abverkaufsstarke Produkte, Konzepte und Ideen für eine erfolgreiche Zusammenarbeit.
        </p>
        {router && router.pathname === '/' ? (
          <Link to="purity" activeClass="active" spy={true} smooth={true} offset={-25} duration={700}>
            <figure id="landing-eco" className="hidden w-56 -mt-16 text-center group lg:block">
              <Image
                className="transition-all duration-500 transform -rotate-10 group-hover:rotate-0"
                src="/certificates/hdm-oekosiegel-header.png"
                alt="HDM - Verantwortung für Mensch und Natur"
                title="Mehr über die den BIO ... logischen Laminatboden erfahren"
                width={300}
                height={300}
              />
            </figure>
          </Link>
        ) : (
          <figure className="hidden w-56 -mt-16 text-center lg:block">
            <Image
              src="/certificates/hdm-oekosiegel-header.png"
              alt="HDM - Verantwortung für Mensch und Natur"
              title="Mehr über die den BIO ... logischen Laminatboden erfahren"
              width={300}
              height={300}
            />
          </figure>
        )}
      </Section>

      {router && router.pathname === '/' && (
        <Sticky enabled={width >= 1024} innerZ="101" activeClass="sticky-nav">
          <nav
            className="justify-around hidden h-24 px-20 py-4 text-lg font-semibold lg:flex"
            aria-label="Produktnavigation in der Desktopansicht."
          >
            <Link
              to="crossflex"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-120}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Dekorpaneele
            </Link>
            <Link
              to="purity"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-25}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Laminatb&ouml;den
            </Link>
            <Link
              to="cerastar"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-40}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Designb&ouml;den
            </Link>
            <Link
              to="massivholztueren"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-40}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Massivholzt&uuml;ren
            </Link>
            <Link
              to="shelves"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-20}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Möbelbauteile
            </Link>
            <Link
              to="strips"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-20}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Leisten
            </Link>
            <Link
              to="accessories"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-20}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              {!process.browser ? 'DAZU-Zubehör' : width < 1280 ? 'Zubehör' : 'DAZU-Zubehör'}
            </Link>
          </nav>
        </Sticky>
      )}
    </header>
  )
}

const Section = styled.section`
  position: relative;

  &::before {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: -2;
    background-color: #f7fafc;
    content: '';
  }

  &::after {
    position: absolute;
    top: 0;
    left: 0;
    z-index: -1;
    width: calc(2.9333 * 90%);
    height: 90%;
    background-image: url(/ci/hdm-bg.svg);
    background-repeat: no-repeat;
    opacity: 0.05;
    content: '';
  }

  @media (min-width: 1024px) {
    position: static;

    &::after {
      left: 6rem;
      width: calc(2.9333 * 100%);
      height: 100%;
    }
  }
`
