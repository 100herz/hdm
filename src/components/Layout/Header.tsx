import NavLink from "next/link"
import { useRouter } from "next/router"
import { Link } from "react-scroll"
import Sticky from "react-stickynode"
import styled from "styled-components"

import { useWindowDimensions } from "@/hooks/useWindowDimensions"

const Section = styled.section`
  position: relative;

  &::before {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: -2;
    background-color: #f7fafc;
    content: "";
  }

  &::after {
    position: absolute;
    top: 0;
    left: 0;
    z-index: -1;
    width: calc(2.9333 * 90%);
    height: 90%;
    background-image: url(/ci/hdm-bg.svg);
    background-repeat: no-repeat;
    opacity: 0.05;
    content: "";
  }

  @media (min-width: 768px) {
    position: static;

    &::after {
      left: 6rem;
      width: calc(2.9333 * 100%);
      height: 100%;
    }
  }
`

type Props = {
  className?: string
}

const Header: React.FC<Props> = ({ className = "" }: Props) => {
  const router = useRouter()
  const { width = 0 } = useWindowDimensions()

  return (
    <header className={`relative overflow-x-hidden ${className}`}>
      <div className="justify-between mt-6 lg:container lg:flex">
        <h1 className="pt-8 mb-10 lg:mb-0">
          <NavLink href="/">
            <a>
              <img className="w-56 mx-auto" src="/ci/hdm-logo.svg" alt="Logo der HDM GmbH" />
              <span className="sr-only">HDM - Holz Design Moers GmbH</span>
            </a>
          </NavLink>
        </h1>

        <Sticky enabled={width < 768} innerZ="101">
          {router && router.pathname === "/" ? (
            <nav className="flex justify-between py-3 font-semibold bg-white lg:bg-transparent lg:text-lg lg:container lg:mb-0 lg:space-x-12 lg:justify-end lg:mt-10 lg:py-0">
              <Link
                to="produkte"
                activeClass="active"
                spy={true}
                smooth={true}
                offset={-120}
                duration={600}
                className="px-4 py-2 my-2 hover:text-hdm"
              >
                Produkte
              </Link>
              <Link
                to="unternehmen"
                activeClass="active"
                spy={true}
                smooth={true}
                offset={-60}
                duration={1000}
                className="px-4 py-2 my-2 hover:text-hdm"
              >
                Unternehmen
              </Link>
              <Link
                to="kontakt"
                activeClass="active"
                spy={true}
                smooth={true}
                duration={1200}
                className="px-4 py-2 my-2 hover:text-hdm"
              >
                Kontakt
              </Link>
            </nav>
          ) : (
            <nav className="flex justify-between py-3 font-semibold bg-white lg:bg-transparent lg:text-lg lg:container lg:mb-0 lg:space-x-12 lg:justify-end lg:mt-10 lg:py-0">
              <NavLink href="/#produkte">
                <a className="px-4 py-2 my-2 hover:text-hdm">Produkte</a>
              </NavLink>
              <NavLink href="/#unternehmen">
                <a className="px-4 py-2 my-2 hover:text-hdm">Unternehmen</a>
              </NavLink>
              <NavLink href="/#kontakt">
                <a className="px-4 py-2 my-2 hover:text-hdm">Kontakt</a>
              </NavLink>
            </nav>
          )}
        </Sticky>
      </div>

      <Section className="container pt-12 pb-20 lg:flex lg:flex-row lg:justify-between lg:pt-32 lg:pb-24">
        <p className="text-xl leading-10 lg:text-3xl lg:max-w-3xl lg:ml-24">
          <strong>Produkt. Idee. Erfolg.</strong> Die Holz Design Moers GmbH bietet ihren Handelspartnern auch in
          Zukunft abverkaufsstarke Produkte, Konzepte und Ideen für eine erfolgreiche Zusammenarbeit.
        </p>
        <Link to="purity" activeClass="active" spy={true} smooth={true} offset={-25} duration={700}>
          <a className="hidden -mt-16 lg:block hover:text-green-600">
            <figure className="text-center group">
              <img
                className="w-56 transition-all duration-500 transform -rotate-10 group-hover:rotate-0"
                src="/certificates/hdm-oekosiegel-header.png"
                alt="HDM - Verantwortung für Mensch und Natur"
                title="Mehr über die den BIO ... logischen Laminatboden erfahren"
              />
            </figure>
          </a>
        </Link>
      </Section>

      {router && router.pathname === "/" && (
        <Sticky enabled={width >= 768} innerZ="101" activeClass="sticky-nav">
          <nav className="justify-around hidden h-24 px-20 py-4 text-lg font-semibold lg:flex">
            <Link
              to="crossflex"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-120}
              duration={600}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Dekorpaneele
            </Link>
            <Link
              to="purity"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-25}
              duration={700}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Laminatb&ouml;den
            </Link>
            <Link
              to="cerastar"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-40}
              duration={900}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Designb&ouml;den
            </Link>
            <Link
              to="massivholztueren"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-40}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Massivholzt&uuml;ren
            </Link>
            <Link
              to="accessories"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-20}
              duration={1100}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              {width >= 768 ? "Regalböden / Möbelbauplatten" : "Böden"}
            </Link>
            <Link
              to="accessories"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-20}
              duration={1100}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              DAZU<sup>&reg;</sup> Zubeh&ouml;r
            </Link>
          </nav>
        </Sticky>
      )}
    </header>
  )
}

export default Header
