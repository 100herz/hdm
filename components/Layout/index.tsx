import { ReactNode } from "react"
import Head from "next/head"

import Header from "@/components/Layout/Header"
import Footer from "@/components/Layout/Footer"

type Props = {
  children?: ReactNode
  title?: string
  description?: string
}

const Layout: React.FC<Props> = ({
  children,
  title = "This is the default title",
  description = "Die Holz Design Moers GmbH bietet ihren Handelspartnern auch in Zukunft abverkaufsstarke Produkte, Konzepte und Ideen fÃ¼r eine erfolgreiche Zusammenarbeit.",
}: Props) => {
  return (
    <div className="antialiased">
      <Head>
        <title>{title}</title>
        <meta charSet="utf-8" />
        <meta name="viewport" content="initial-scale=1.0, width=device-width" />
        <meta name="description" content={description} />
        <link rel="shortcut icon" href="/ci/hdm-logo.svg" type="image/svg+xml" sizes="any" />
        <link
          rel="stylesheet"
          href="https://fonts.googleapis.com/css2?family=Montserrat:ital,wght@0,200;0,400;0,700;1,200;1,400;1,700&display=swap"
        />
      </Head>

      <Header />

      <main>{children}</main>

      <Footer />
    </div>
  )
}

export default Layout
