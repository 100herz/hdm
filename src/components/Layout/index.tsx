import { ReactNode, useEffect } from "react"
import Head from "next/head"

import Header from "@/components/Layout/Header"
import Footer from "@/components/Layout/Footer"

import { initFocusRingOnlyOnTab } from "@/utils/accessibility"

type Props = {
  children?: ReactNode
  title?: string
  description?: string
}

const Layout: React.FC<Props> = ({
  children,
  title = "This is the default title",
  description = "Die Holz Design Moers GmbH bietet ihren Handelspartnern auch in Zukunft abverkaufsstarke Produkte, Konzepte und Ideen fÃ¼r eine erfolgreiche Zusammenarbeit.",
}: Props) => {
  useEffect(() => {
    initFocusRingOnlyOnTab()
  }, [])

  return (
    <div className="antialiased">
      <Head>
        <title>{title}</title>
        <meta charSet="utf-8" />
        <meta name="viewport" content="initial-scale=1.0, width=device-width" />
        <meta name="description" content={description} />
        <meta name="robots" content="noindex, nofollow" />
        <link rel="shortcut icon" href="/ci/hdm-bg.svg" type="image/svg+xml" sizes="any" />
        <link
          href="https://fonts.googleapis.com/css2?family=Open+Sans:wght@400;600;700&display=swap"
          rel="stylesheet"
        />
      </Head>

      <Header />

      <main>{children}</main>

      <Footer />
    </div>
  )
}

export default Layout
