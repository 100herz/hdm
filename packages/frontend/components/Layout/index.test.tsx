import { render } from '@/test/test-utils'

import Layout from '.'

describe('Layout component', () => {
  it('matches snapshot', () => {
    const { container, getByText } = render(<Layout />)
    expect(getByText(/Produkt. Idee. Erfolg./i)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})

// TODO: Test the `links` and `utils`
