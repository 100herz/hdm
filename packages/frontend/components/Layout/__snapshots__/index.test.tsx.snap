// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component matches snapshot 1`] = `
.emotion-0 {
  position: relative;
}

.emotion-0::before {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: -2;
  background-color: #f7fafc;
  content: '';
}

.emotion-0::after {
  position: absolute;
  top: 0;
  left: 0;
  z-index: -1;
  width: calc(2.9333 * 90%);
  height: 90%;
  background-image: url(/ci/hdm-bg.svg);
  background-repeat: no-repeat;
  opacity: 0.05;
  content: '';
}

@media (min-width: 1024px) {
  .emotion-0 {
    position: static;
  }

  .emotion-0::after {
    left: 6rem;
    width: calc(2.9333 * 100%);
    height: 100%;
  }
}

.emotion-2 {
  position: relative;
}

@media (min-width: 1024px) {
  .emotion-2 {
    background-image: linear-gradient(rgba(255, 255, 255, 0.95), rgba(255, 255, 255, 0.95)),url(/ci/hdm-bg.svg);
    background-repeat: no-repeat;
    -webkit-background-size: 100% auto;
    background-size: 100% auto;
  }
}

.emotion-4 {
  font-size: 1.125rem;
  font-style: normal;
  line-height: 2.5rem;
}

<div
  class="antialiased"
>
  <header
    class="relative overflow-x-hidden "
  >
    <div
      class="justify-between mt-6 lg:container lg:flex"
    >
      <h1
        class="pt-8 mb-10 lg:mb-0"
        data-testid="main-logo"
      >
        <a
          aria-label="HDM - Holz Design Moers GmbH"
          class="block w-56 mx-auto"
          href="/"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjI0IiBoZWlnaHQ9IjU3IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <img
              alt="Logo der HDM GmbH"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
        </a>
      </h1>
      <div
        class="sticky-outer-wrapper"
      >
        <div
          class="sticky-inner-wrapper "
          style="position: relative; top: 0px; z-index: 101;"
        >
          <nav
            aria-label="Hauptnavigation nicht auf der Startseite aber in der mobilen Ansicht."
            class="flex justify-between py-3 font-semibold bg-white lg:bg-transparent lg:text-lg lg:container lg:mb-0 lg:space-x-12 lg:justify-end lg:mt-10 lg:py-0"
          >
            <a
              class="px-4 py-2 my-2 hover:text-hdm"
              href="/#produkte"
            >
              Produkte
            </a>
            <a
              class="px-4 py-2 my-2 hover:text-hdm"
              href="/#unternehmen"
            >
              Unternehmen
            </a>
            <a
              class="px-4 py-2 my-2 hover:text-hdm"
              href="/#kontakt"
            >
              Kontakt
            </a>
          </nav>
        </div>
      </div>
    </div>
    <section
      class="container pt-12 pb-20 lg:flex lg:flex-row lg:justify-between lg:pt-32 lg:pb-24 emotion-0 emotion-1"
    >
      <p
        class="text-xl leading-10 lg:text-3xl lg:max-w-3xl lg:ml-24"
      >
        <strong>
          Produkt. Idee. Erfolg.
        </strong>
         Die Holz Design Moers GmbH bietet ihren Handelspartnern auch in Zukunft abverkaufsstarke Produkte, Konzepte und Ideen für eine erfolgreiche Zusammenarbeit.
      </p>
      <figure
        class="hidden w-56 -mt-16 text-center lg:block"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzAwIiBoZWlnaHQ9IjMwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <img
            alt="HDM - Verantwortung für Mensch und Natur"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            title="Mehr über die den BIO ... logischen Laminatboden erfahren"
          />
        </div>
      </figure>
    </section>
  </header>
  <main />
  <footer
    class="lg:mt-40 pb-10  emotion-2 emotion-3"
    id="kontakt"
  >
    <section
      class="container lg:flex lg:justify-between lg:space-x-6"
    >
      <h2
        class="py-16 text-4xl font-bold leading-10 lg:pt-0"
      >
        Nehmen Sie 
        <br />
        Kontakt mit 
        <br />
        uns auf
      </h2>
      <address
        class="emotion-4 emotion-5"
      >
        Holz Design Moers GmbH 
        <br />
        Am Schürmannshütt 23 
        <br />
        47441 Moers
      </address>
      <address
        class="mt-6 lg:mt-0 emotion-4 emotion-5"
      >
        Telefon:
         
        <a
          class="underline hover:text-hdm-dark"
          href="tel:+4928411401101"
        >
          +49 2841 1401 101
        </a>
        <br />
        Telefax: +49 2841 1401 180 
        <br />
        E-Mail:
         
        <a
          class="underline hover:text-hdm-dark"
          href="mailto:info@hdm.gmbh"
        >
          info@hdm.gmbh
        </a>
      </address>
      <div
        class="mt-24 lg:mt-0"
      >
        <a
          aria-label="Zum Seitenanfang springen"
          class="w-48"
          href="/"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTkyIiBoZWlnaHQ9IjQ5IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <img
              alt="Logo der HDM GmbH"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
        </a>
      </div>
    </section>
    <section
      class="container mt-6 text-lg leading-10 lg:flex lg:justify-center"
    >
      <p>
        © 2020 Holz Design Moers GmbH
      </p>
      <nav
        class="flex flex-col mt-2 leading-8 lg:mt-0 lg:flex-row"
      >
        <a
          class="underline lg:ml-8 hover:text-hdm-dark"
          href="/downloads"
        >
          Downloads
        </a>
        <a
          class="underline lg:ml-8 hover:text-hdm-dark"
          href="/impressum"
        >
          Impressum
        </a>
        <a
          class="underline lg:ml-8 hover:text-hdm-dark"
          href="/datenschutz"
        >
          Datenschutz
        </a>
      </nav>
    </section>
    <div
      class="container my-10 text-right lg:text-center"
    >
      <button
        aria-label="Go to top"
        data-testid="back-to-top-button"
        type="button"
      >
        <svg
          class="inline-block w-6 transform -rotate-90"
          viewBox="0 0 24.242 45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M126.031,21.266,105.252.486A1.732,1.732,0,0,0,102.8,2.935L122.359,22.49,102.8,42.045a1.731,1.731,0,0,0,2.448,2.448l20.779-20.779A1.732,1.732,0,0,0,126.031,21.266Z"
            fill="#0084b0"
            transform="translate(-102.297 0)"
          />
        </svg>
      </button>
    </div>
  </footer>
</div>
`;
